internal/server/utils/score_parser.go

```diff
@@ -0,0 +1,19 @@
+// utils/score_parser.go
+package utils
+
+import (
+	"math"
+	"strconv"
+)
+
+// ParseScore converts a string to a float64, handling special cases like -inf and +inf.
+func ParseScore(input string) (float64, error) {
+	switch input {
+	case "-inf":
+		return math.Inf(-1), nil
+	case "+inf":
+		return math.Inf(1), nil
+	default:
+		return strconv.ParseFloat(input, 64)
+	}
+}
```

Comment:
Please confirm is strconv can be used instead of this util function

Reply:
Resolved